tests/lib/rules/anchor-label.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Rule to enforce descriptive anchor label
+ * @author Taylor Seamans
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require('../../../lib/rules/anchor-label'),
+  RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester({
+  parserOptions: { ecmaFeatures: { jsx: true } },
+});
+ruleTester.run('anchor-label', rule, {
+  valid: [
+    '<Anchor label="Grommet docs" href="https://grommet.io" />',
+    '<Anchor href="https://grommet.io">Grommet docs</Anchor>',
+  ],
+  invalid: [
+    {
+      code: '<Anchor label="here" href="/some-page" />',
+      errors: [
+        {
+          messageId: 'anchor-label',
+        },
+      ],
+    },
+    {
+      code: '<Anchor label="Click here" href="/some-page" />',
+      errors: [
+        {
+          messageId: 'anchor-label',
+        },
+      ],
+    },
+    {
+      code: '<Anchor href="/some-page">Click here</Anchor>',
```

Comment:
What will happen for something like `<Anchor><Text weight="bold">Click here</Text></Anchor>`?

Reply:
Great point -- I will try looping down the children of children.

Reply:
I wonder if the children could get complex and we should give up at some point. Just `label` or direct string children will definitely be the 80% case. We might want to just say nothing for anything else.

Reply:
I am good with that. I agree, there would be complex child cases that we can't necessarily account for, which is why I steered for the simple/likely cases at first.

Another option is to leave as is for now and could always enhance the rule in the future if we wanted.

Reply:
Can we add a test, to ensure we handle an example like this appropriately?

Reply:
If we are leaving the rule as is, then `<Anchor><Text weight="bold">Click here</Text></Anchor>` would go into the "correct" section. 

If we update the rule to try to account for nested children, I can put it in the "incorrect" section.

What direction do you want to head? Trying to loop through children or just leaving as is (label + direct children)?

Reply:
Is there the possibility of "no opinion" section?

Reply:
That would technically be "correct" meaning it wouldn't be flagged -- but if we don't want to document it as a "correct case" then we should just omit it from the test code.

Reply:
Thanks for teaching me the eslint plugin details.